docs/en/reference/basic-mapping.rst

```diff
@@ -209,27 +209,39 @@ built-in mapping types:
 -  ``decimal``: Type that maps a SQL DECIMAL to a PHP string.
 -  ``date``: Type that maps a SQL DATETIME to a PHP DateTime
    object.
+-  ``date_immutable``: Type that maps a SQL DATETIME to a PHP DateTimeImmutable
+   object.
 -  ``time``: Type that maps a SQL TIME to a PHP DateTime object.
+-  ``time_immutable``: Type that maps a SQL TIME to a PHP DateTimeImmutable object.
 -  ``datetime``: Type that maps a SQL DATETIME/TIMESTAMP to a PHP
    DateTime object.
+-  ``datetime_immutable``: Type that maps a SQL DATETIME/TIMESTAMP to a PHP
+   DateTimeImmutable object.
 -  ``datetimetz``: Type that maps a SQL DATETIME/TIMESTAMP to a PHP
    DateTime object with timezone.
+-  ``datetimetz_immutable``: Type that maps a SQL DATETIME/TIMESTAMP to a PHP
+   DateTimeImmutable object with timezone.
 -  ``text``: Type that maps a SQL CLOB to a PHP string.
 -  ``object``: Type that maps a SQL CLOB to a PHP object using
    ``serialize()`` and ``unserialize()``
 -  ``array``: Type that maps a SQL CLOB to a PHP array using
    ``serialize()`` and ``unserialize()``
--  ``simple_array``: Type that maps a SQL CLOB to a PHP array using
+-  ``simple_array``: Type that maps a SQL CLOB to a one-dimensional PHP array using
    ``implode()`` and ``explode()``, with a comma as delimiter. *IMPORTANT*
    Only use this type if you are sure that your values cannot contain a ",".
 -  ``json_array``: Type that maps a SQL CLOB to a PHP array using
-   ``json_encode()`` and ``json_decode()``
+   ``json_encode()`` and ``json_decode()``. This one was deprecated in favor
+   of ``json`` type.
+-  ``json``: Type that maps a SQL CLOB to a PHP array using
+   ``json_encode()`` and ``json_decode()``. An empty value is correctly represented as ``null``
 -  ``float``: Type that maps a SQL Float (Double Precision) to a
    PHP double. *IMPORTANT*: Works only with locale settings that use
    decimal points as separator.
 -  ``guid``: Type that maps a database GUID/UUID to a PHP string. Defaults to
    varchar but uses a specific type if the platform supports it.
 -  ``blob``: Type that maps a SQL BLOB to a PHP resource stream
+-  ``binary``: Type that maps a SQL binary to a PHP resource stream
+-  ``dateinterval``: Type that maps an interval to a PHP DateInterval object
```

Comment:
Should this type reside somewhere closer to the rest of the date-related ones?

Reply:
Makes sense to me.